### ISSUE TITLE
Paddle CheckPoint的保存问题

### ISSUE THREAD

**Issue by larryhu76** [PaddlePaddle/PaddleSeg#289]
在Tensorflow中，keras.callbacks.ModelCheckpoint(  monitor='loss',）回调中monitor参数可以指定模型保存的时机，比如loss表现更好的时候才保存模型；
但在Paddle中，只能按照TRAIN.SNAPSHOT_EPOCH 这个参数来保存模型；
请问是否有类似TF中的模型保存方案？

**Comment by wuyefeilin** [comment]
如果开启边训练边评估的话，我们会保存miou最好的模型到best_model中

**Comment by larryhu76** [comment]
哪个参数可以开启边训练边评估的？best_model的目录需要单独配置吗？

**Comment by wuyefeilin** [comment]
python train.py --do_eval  开启边训练边评估。best_model不需要单独配置

**Comment by larryhu76** [comment]
开启 --do_eval模式后，是多久保存一次的？TRAIN.SNAPSHOT_EPOCH这个参数还有效吗？

**Comment by wuyefeilin** [comment]
TRAIN.SNAPSHOT_EPOCH 按照这个间隔进行保存

**Comment by larryhu76** [comment]
“我们会保存miou最好的模型到best_model中”，这个best_model在哪个目录下可以看到？是在整个训练结束后再生成的吗？

**Comment by wuyefeilin** [comment]
best_mode 在你模型的保存路径中，只要有模型开始保存就会生成best_model, 然后会根据每次评估的miou进行评估

**Comment by larryhu76** [comment]
我打开了 --do_eval，SNAPSHOT_EPOCH 设置为20,目前一训练了100多次，每20 个 Epoch可以看到对验证集的评估结果 和 保存的模型，但是没有看到 best_model保存下来。PaddlePaddle 框架是1.5.1, Python 3.7,是不是 Paddle框架版本太低了,还是哪里没配置好？

**Comment by wuyefeilin** [comment]
你paddleseg用的版本什么版本

**Comment by larryhu76** [comment]
是V0.3.0

**Comment by larryhu76** [comment]
是V0.3.0

**Comment by wuyefeilin** [comment]
用最新版本

**Comment by larryhu76** [comment]
升级到V0.50,可以保存到 best_model中，问题解决。
